google/cloud/storage/_helpers.py

```diff
@@ -449,6 +466,24 @@ def _convert_to_timestamp(value):
     return mtime
 
 
+def _add_etag_match_headers(headers, **match_parameters):
+    """Add generation match parameters into the given parameters list.
+
+    :type headers: dict
+    :param headers: Headers dict.
+
+    :type match_parameters: dict
+    :param match_parameters: if*etag*match parameters to add.
+    """
+    for snakecase_name, header_name in _ETAG_MATCH_PARAMETERS:
+        value = match_parameters.get(snakecase_name)
+
+        if value is not None:
+            if isinstance(value, str):
+                value = [value]
+            headers[header_name] = ", ".join(value)
```

Comment:
It seems to me (though I'm not a spec expert) that the GCS API isn't following spec putting quotes around the etag values. E.g. `If-None-Match: COKaz4vVzfECEAE=` works as intended but `If-None-Match: "COKaz4vVzfECEAE="` does not. That said, the response header is also not quoted as it apparently should be `etag: COKaz4vVzfECEAE=` vs expected `etag: "COKaz4vVzfECEAE="`. Referencing [this](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/ETag) and [this](https://stackoverflow.com/questions/6719214/syntax-for-etag).

Reply:
Hmm, maybe its just the display layers or something (using Chrome dev tools and curl). Wikipedia does the same thing.

Reply:
microsoft.com has it quoted 🤔

Reply:
RFC 7232 is the actual authoritative spec for conditional HTTP requests: [`If-Match`](https://datatracker.ietf.org/doc/html/rfc7232#section-3.1) and [`If-None-Match`](https://datatracker.ietf.org/doc/html/rfc7232#section-3.2).  The [Collected ABNF appendix](https://datatracker.ietf.org/doc/html/rfc7232#appendix-C) does indeed seem to require the double quote.

The [GCS docs for the ETag header](https://cloud.google.com/storage/docs/json_api/v1/parameters#etag) show the quotes as well.

Reply:
Yep, all the examples in the spec are also quoted. Fun times. Definitely out of scope for this PR. Just wanted to call it out. I would think the etag itself should have the quotes (i.e. `blob.etag = '"Csdlei="'`), so then this code would still function correctly. Of course getting etags quoted would be a far larger change spanning the backend, possibly ESF, and probably a fair amount of Hyrum's law.

It just stood out to me because I ran into a similar problem implementing the UI for BigQuery, which also doesn't follow the etag spec. In the end it meant we couldn't cache tables in the UI. But we also couldn't change the API without risking breaking clients. I'll leave it up to you if you want to file an internal ticket for that.

Reply:
Effectively we're stuck with the as-implemented back-end.  @frankyn, @andrewsg, @tswast I'll leave it to y'all to open a ticket on the back-end (if one isn't there already) for the out-of-spec behavior.